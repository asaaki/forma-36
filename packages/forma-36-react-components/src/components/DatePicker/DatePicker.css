@import 'settings/colors';
@import 'settings/typography';
@import 'settings/transitions';

.DatePicker {
  display: inline-block;
  font-family: var(--font-stack-primary);
  font-size: calc(1rem * (14 / var(--font-base-default)));
}

.DatePickerCalendar {
  font-family: var(--font-stack-primary);
  font-size: 1rem;
}

.DatePickerCalendar button,
.DatePicker button {
  border: none;
  background: none;
  font-size: 1rem;
  appearance: none;
}
