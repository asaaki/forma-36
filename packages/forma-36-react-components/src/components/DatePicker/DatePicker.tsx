import React, { Component } from 'react';
import ReactDatePicker from 'react-date-picker/dist/entry.nostyle';
import cn from 'classnames';

const styles = require('./DatePicker.css');

export type DatePickerProps = {
  /**
   * Class names to be appended to the className prop of the component
   */
  className?: string;
  /**
   * Child nodes to be rendered in the component
   */
  children?: React.ReactNode;
  /**
   * An ID used for testing purposes applied as a data attribute (data-test-id)
   */
  testId?: string;
} & typeof defaultProps;

const defaultProps = {
  testId: 'cf-ui-date-picker',
};

export class DatePicker extends Component<DatePickerProps> {
  static defaultProps = defaultProps;

  state = {
    isOpen: true,
  };

  handleClose = () => {
    this.setState({ isOpen: false });
  };

  render() {
    const { className, children, testId, ...otherProps } = this.props;
    const classNames = cn(styles.DatePicker, className);

    return (
      <ReactDatePicker
        onCalendarClose={this.handleClose}
        className={classNames}
        calendarClassName={styles.DatePickerCalendar}
        date-text-id={testId}
        isOpen={this.state.isOpen}
        {...otherProps}
      />
    );
  }
}

export default DatePicker;
