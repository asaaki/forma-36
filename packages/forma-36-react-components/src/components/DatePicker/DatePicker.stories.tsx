import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';

import DatePicker from './DatePicker';

storiesOf('Components|DatePicker', module)
  .addParameters({
    propTypes: DatePicker['__docgenInfo'],
  })
  .add('default', () => (
    <DatePicker className={text('className', '')}>DatePicker</DatePicker>
  ));
