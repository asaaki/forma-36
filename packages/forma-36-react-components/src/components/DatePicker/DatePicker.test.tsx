import React from 'react';
import { shallow } from 'enzyme';
import DatePicker from './DatePicker';

it('renders the component', () => {
  const output = shallow(<DatePicker>DatePicker</DatePicker>);

  expect(output).toMatchSnapshot();
});

it('renders the component with an additional class name', () => {
  const output = shallow(
    <DatePicker className="my-extra-class">DatePicker</DatePicker>,
  );

  expect(output).toMatchSnapshot();
});
