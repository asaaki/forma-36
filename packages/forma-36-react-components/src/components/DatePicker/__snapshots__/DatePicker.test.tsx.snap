// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="DatePicker"
  data-test-id="cf-ui-date-picker"
>
  <DatePicker
    calendarIcon={
      <svg
        height="19"
        viewBox="0 0 19 19"
        width="19"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          stroke="black"
          strokeWidth="2"
        >
          <rect
            fill="none"
            height="15"
            width="15"
            x="2"
            y="2"
          />
          <line
            x1="6"
            x2="6"
            y1="0"
            y2="4"
          />
          <line
            x1="13"
            x2="13"
            y1="0"
            y2="4"
          />
        </g>
      </svg>
    }
    clearIcon={
      <svg
        height="19"
        viewBox="0 0 19 19"
        width="19"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          stroke="black"
          strokeWidth="2"
        >
          <line
            x1="4"
            x2="15"
            y1="4"
            y2="15"
          />
          <line
            x1="15"
            x2="4"
            y1="4"
            y2="15"
          />
        </g>
      </svg>
    }
    isOpen={true}
    returnValue="start"
  />
</div>
`;

exports[`renders the component with an additional class name 1`] = `
<div
  className="DatePicker my-extra-class"
  data-test-id="cf-ui-date-picker"
>
  <DatePicker
    calendarIcon={
      <svg
        height="19"
        viewBox="0 0 19 19"
        width="19"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          stroke="black"
          strokeWidth="2"
        >
          <rect
            fill="none"
            height="15"
            width="15"
            x="2"
            y="2"
          />
          <line
            x1="6"
            x2="6"
            y1="0"
            y2="4"
          />
          <line
            x1="13"
            x2="13"
            y1="0"
            y2="4"
          />
        </g>
      </svg>
    }
    clearIcon={
      <svg
        height="19"
        viewBox="0 0 19 19"
        width="19"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          stroke="black"
          strokeWidth="2"
        >
          <line
            x1="4"
            x2="15"
            y1="4"
            y2="15"
          />
          <line
            x1="15"
            x2="4"
            y1="4"
            y2="15"
          />
        </g>
      </svg>
    }
    isOpen={true}
    returnValue="start"
  />
</div>
`;
